components/spi_flash/flash_ops.c

```diff
@@ -127,121 +128,147 @@ esp_err_t IRAM_ATTR spi_flash_erase_range(uint32_t start_addr, uint32_t size)
     return spi_flash_translate_rc(rc);
 }
 
-esp_err_t IRAM_ATTR spi_flash_write(size_t dest_addr, const void *src, size_t size)
+esp_err_t IRAM_ATTR spi_flash_write(size_t dst, const void *srcv, size_t size)
 {
-    // Destination alignment is also checked in ROM code, but we can give
-    // better error code here
-    // TODO: add handling of unaligned destinations
-    uint8_t *temp_write_buf = NULL;
-    uint8_t pad_head = 0;
-    uint8_t pad_end = 0;
-    SpiFlashOpResult rc;
-    // Out of bound writes are checked in ROM code, but we can give better
-    // error code here
-    if (dest_addr + size > g_rom_flashchip.chip_size) {
-        return ESP_ERR_INVALID_SIZE;
-    }
-
-    while(size >= 1024) {
-        // max need pad byte num for 1024 is 4
-        temp_write_buf = (uint8_t*)malloc(1024 + 4);
-        if(temp_write_buf == NULL) {
-            return ESP_ERR_NO_MEM;
-        }
+    if (size == 0) return ESP_OK;
 
-        if(dest_addr%4 != 0) {
-            pad_head = dest_addr%4;
-            pad_end = 4 - pad_head;
-        }
-        memset(temp_write_buf,0xFF,pad_head);
-        memcpy(temp_write_buf + pad_head ,src,1024);
-        memset(temp_write_buf + pad_head + 1024, 0xFF,pad_end);
-        COUNTER_START();
-        spi_flash_disable_interrupts_caches_and_other_cpu();
-        rc = spi_flash_unlock();
-        if (rc == SPI_FLASH_RESULT_OK) {
-            rc = SPIWrite((uint32_t) (dest_addr - pad_head), (const uint32_t*) temp_write_buf, (int32_t) (1024 + pad_head + pad_end));
-            COUNTER_ADD_BYTES(write, 1024 + pad_head + pad_end);
-        }
-        COUNTER_STOP(write);
-        spi_flash_enable_interrupts_caches_and_other_cpu();  
-        if(rc != ESP_OK) {
-            free(temp_write_buf);
-            temp_write_buf = NULL;
-            return spi_flash_translate_rc(rc);
-        }
+    // Out of bound reads are checked in ROM code, but we can give better
+    // error code here
+    if (dst + size > g_rom_flashchip.chip_size) return ESP_ERR_INVALID_SIZE;
 
-        free(temp_write_buf);
-        temp_write_buf = NULL;
-        size -= 1024;
-        dest_addr += 1024;
-        src = (uint8_t*)src + 1024;
+    SpiFlashOpResult rc = SPI_FLASH_RESULT_OK;
+    COUNTER_START();
+    spi_flash_disable_interrupts_caches_and_other_cpu();
+    const char *srcc = (const char *) srcv;
```

Comment:
Hi rojer，I think if this api want to write data in Dcache to falsh,and the pointer point to memory of Dcache, but spi_flash_disable_interrupts_caches_and_other_cpu() closed cache,so when you call memcpy(((uint8_t *) &t) + (dst - left_off), srcc, left_size); I think it will crash,I'm not sure,you can test it.